js/reference.js

```diff
@@ -6595,3 +6618,126 @@
+     */
+    function Prefill (reference, prefillObject, prefillAnnotation) {
+        verify(prefillObject, "'prefillObject' must be defined");
+
+        var self = this;
+        this._reference = reference;
+        this._prefillObject = prefillObject;
+        this._prefillAnnotation = prefillAnnotation;
+
+        // ignore the fks that are simple and their constituent column is system col
+        var nonSystemColumnFks = this._reference.table.foreignKeys.all().filter(function(fk) {
+            return !(fk.simple && module._systemColumns.indexOf(fk.colset.columns[0]) !== -1);
+        });
+
+        // set of foreignkey columns (they might be overlapping so we're not using array)
+        var fkCols = {};
+        nonSystemColumnFks.forEach(function(fk) {
+            fk.colset.columns.forEach(function(col) {
+                fkCols[col] = true;
+            });
+        });
+
+        // the following are ERMrest.ForeignKeyPseudoColumn
+        this.mainColumn;
+        this.leafColumn;
+
+         // NOTE: I think this can be done differently. Leaving this alone for now for the initial implementation
+        reference.columns.forEach(function(column) {
+            // column should be a foreignkey pseudo column
+            if (!column.isForeignKey) return;
+
+            nonSystemColumnFks.forEach(function(fk) {
+                // column and foreign key `.name` property is a hash value
+                if (column.name === fk.name) {
+                    if (prefillObject.fkColumnNames.indexOf(column.name) !== -1) {
+                        self.mainColumn = column;
+                    } else {
+                        // TODO: what if there are multiple "other" fk columns?
+                        if (self.leafColumn) {
+                            // throw error saying "more than 1 foreign key column available"
+                            // OR inspect key data and if there is a key with 2 foreign key columns (and 1 is the main column), those are our columns for the association
+                        }
+
+                        self.leafColumn = column;
+                    }
+                }
+            });
+        });
+
+        this.isAssociation = (nonSystemColumnFks.length === 2 && this.mainColumn && this.leafColumn);
+
+        this.isUnique = false;
+
+        // to calculate isUnique
+        //   - One of the keys should contain the main and leaf foreign key columns
+        var tempKeys = reference.table.keys.all().filter(function(key) {
+            var keyCols = key.colset.columns;
+            return !(keyCols.length == 1 && (module._serialTypes.indexOf(keyCols[0].type.name) != -1 ||  module._systemColumns.indexOf(keyCols[0].name) != -1) && !(keyCols[0] in fkCols));
+        });
+
+        // if we determine there is a key with 2 foreign key columns in it, should we assume that's the main/leaf columns for the association
+        //   even if there are more than 2 foreign key columns?
```

Comment:
This comment goes along with the comment I made above with `OR` at the start of it. If we don't want to be "smart" and base `isAssociation` only on having 2 FKs, then this comment and the one above don't matter